src/game/GameWorldLoader.ts

```diff
@@ -0,0 +1,300 @@
+    const stats = statsFromData(gameWorld.stats)
+    const stateExtensions = stateExtensionsFromData(
+        gameWorld.worldStateModifiers,
+    )
+    return new BasicGame<Params>([...cards], stats, defaultParams, {
+        tickModifiers: [...events, ...stateExtensions, parameterCaps],
+        random,
+    })
+}
+
+/**
+ * Creates a StateModifier representing an event using data from GameWorld
+ *
+ * @param event The event data
+ * @param card The card to which the event should be linked
+ * @param random The random function used for the probability check
+ * @returns An event trigger in the form of a StateModifier
+ */
+function eventFromData(
+    event: GameWorld['events'][number],
+    card: Card<Params>,
+    random: () => number,
+): StateModifier<Params> {
+    const paramQueries = event.isAvailableWhen.map(worldQueryToParamQuery)
+    return (state) => {
+        const noPreviousCardSetter = !state.card
+        const propabilityHit = random() <= event.probability
+        const shouldExecute =
+            noPreviousCardSetter &&
+            propabilityHit &&
+            hasMatchingParamQuery(state.params, paramQueries)
+        return shouldExecute
+            ? {
+                  ...state,
+                  card: card,
+              }
+            : state
+    }
+}
+
+/**
+ * Creates a list of Stat<Params> from GameWorld data
+ *
+ * @param stats The stats data
+ * @returns A list of Stats
+ */
+function statsFromData(stats: GameWorld['stats']): Stat<Params>[] {
+    return stats.map<Stat<Params>>((stat) => ({
+        ...stat,
+        getValue: ({ params }) => params.vars[stat.id] ?? 0,
+    }))
+}
+
+/**
+ * Creates the standard parameters caps used for a standard GameWorld
+ *
+ * @param stats The stats data
+ * @returns A stat modifier which caps all the vars matched to ids in stats to [0, 100]
+ */
+function parameterCapsFromStats(stats: GameWorld['stats']) {
+    const statVarIds = stats.map((stat) => stat.id)
+    return createParameterCap(statVarIds, 0, 100)
+}
+
+/**
+ * Creates a Card<Params> from regular card data or event card data
+ *
+ * @param data The card data
+ * @param defaultParams The default parameters to use for state reset
+ * @returns A runtime model of a card
+ */
+function cardFromData(
+    data: GameWorld['cards'][number] | GameWorld['eventCards'][string],
+    defaultParams: Params,
+): Card<Params> {
+    const paramQueries = ('isAvailableWhen' in data
+        ? data.isAvailableWhen
+        : []
+    ).map(worldQueryToParamQuery)
+    return {
+        image: data.image,
+        title: data.title,
+        text: data.text,
+        location: data.location,
+        match: (s) => hasMatchingParamQuery(s.params, paramQueries),
+        weight: data.weight,
+        actions: {
+            left: actionFromData(data.actions.left, defaultParams, 'No'),
+            right: actionFromData(data.actions.right, defaultParams, 'Yes'),
```

Comment:
Ah, you fixed the bug mentioned in https://github.com/Greenheart/swipeforfuture.com/pull/156#discussion_r520878358. Nice!

Reply:
There hare two separate implementations here sadly. Due to the internals of Scenario still working with `state`. Thus this does not fix the previously mentioned issue but I have made a fix for the issue in this branch,